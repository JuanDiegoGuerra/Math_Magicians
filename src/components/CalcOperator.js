import PropTypes from 'prop-types';

const CalcOperator = ({ operator, handleClick }) => {
  const handleKeyDown = (event) => {
    if (event.key === 'Enter') {
      handleClick(operator);
    }
  };

  const isOrange = () => (operator === 'AC' || operator === '%' || operator === '+/-' || operator === '.');

  return (
    <div role="button" className={isOrange() ? 'calc-number' : 'calc-number calc-op'} onClick={() => { handleClick(operator); }} onKeyDown={handleKeyDown} tabIndex={0}>{operator}</div>
  );
};

export default CalcOperator;

CalcOperator.propTypes = {
  operator: PropTypes.string.isRequired,
  handleClick: PropTypes.func.isRequired,
};
