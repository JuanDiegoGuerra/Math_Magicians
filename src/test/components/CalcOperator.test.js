import React from 'react';
import { render, screen } from '@testing-library/react';
import renderer from 'react-test-renderer';
import userEvent from '@testing-library/user-event';
import '@testing-library/jest-dom';
import CalcOperator from '../../components/CalcOperator';

describe('The CalcOperator component', () => {
  const handleClick = jest.fn();

  it('Renders orrectly', () => {
    render(<CalcOperator operator="+/-" handleClick={handleClick} />);
    expect(screen.getByRole('button')).toBeInTheDocument();
  });

  it('Matches the snapShot', () => {
    const tree = renderer.create(<CalcOperator operator="AC" handleClick={handleClick} />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('Calls the eventHandler when clicked', () => {
    render(<CalcOperator operator="%" handleClick={handleClick} />);
    const operatorBtn = screen.getByRole('button', {
      name: '%',
    });
    userEvent.click(operatorBtn);
    expect(handleClick).toHaveBeenCalledTimes(1);
  });
});
