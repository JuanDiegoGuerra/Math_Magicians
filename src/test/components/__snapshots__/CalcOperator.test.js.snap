// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The CalcOperator component Matches the snapShot 1`] = `
<div
  className="calc-number"
  onClick={[Function]}
  onKeyDown={[Function]}
  role="button"
  tabIndex={0}
>
  AC
</div>
`;
